openedx/core/djangoapps/content_libraries/views_collections.py

```diff
@@ -0,0 +1,184 @@
+"""
+Collections API Views
+"""
+
+from __future__ import annotations
+
+from rest_framework.decorators import action
+from rest_framework.response import Response
+from rest_framework.viewsets import ModelViewSet
+from rest_framework.status import HTTP_405_METHOD_NOT_ALLOWED
+
+from opaque_keys.edx.locator import LibraryLocatorV2
+
+from openedx.core.djangoapps.content_libraries import api, permissions
+from openedx.core.djangoapps.content_libraries.models import ContentLibrary
+from openedx.core.djangoapps.content_libraries.views import convert_exceptions
+from openedx.core.djangoapps.content_libraries.serializers import (
+    ContentLibraryCollectionSerializer,
+    ContentLibraryCollectionComponentsUpdateSerializer,
+    ContentLibraryCollectionCreateOrUpdateSerializer,
+)
+
+from openedx_learning.api.authoring_models import Collection
+from openedx_learning.api import authoring as authoring_api
+
+
+class LibraryCollectionsView(ModelViewSet):
+    """
+    Views to get, create and update Library Collections.
+    """
+
+    serializer_class = ContentLibraryCollectionSerializer
+
+    def _verify_and_fetch_library_collection(
+        self,
+        library_key,
+        collection_id,
+        user,
+        permission,
+    ) -> tuple[ContentLibrary, Collection]:
+        """
+        Verify that the collection belongs to the library and the user has the correct permissions.
+
+        This method may raise exceptions; these are handled by the @convert_exceptions wrapper on the views.
+        """
+        library_obj = api.require_permission_for_library_key(library_key, user, permission)
+        collection = None
+        if library_obj.learning_package_id:
+            collection = authoring_api.get_collections(
+                library_obj.learning_package_id
+            ).filter(id=collection_id).first()
+        if not collection:
+            raise api.ContentLibraryCollectionNotFound
+        return library_obj, collection
+
+    @convert_exceptions
+    def retrieve(self, request, *args, **kwargs):
+        """
+        Retrieve the Content Library Collection
+        """
+        lib_key_str = kwargs.pop("lib_key_str")
+        library_key = LibraryLocatorV2.from_string(lib_key_str)
+        pk = kwargs.pop("pk")
```

Comment:
@bradenmacdonald re the [`collection.slug` addition](https://github.com/openedx/edx-platform/pull/35321#issuecomment-2323788286) -- should we use the `collection.slug` or the full LibraryCollectionLocator in these view URLs instead of the `collection.pk`?

Reply:
I would just use the collection slug in the URL. Basically I think the only place we should use the opaque keys (`LibraryCollectionKey`) is for tagging since it requires an OpaqueKey, and everywhere else (if feasible) just use the library ID/key and the collection slug as separate parameters. Then we don't have to deal with opaquekeys anywhere else.